pkg/driver/distro/distro.go

```diff
@@ -179,6 +199,25 @@ func Build(ctx context.Context,
 	if err != nil {
 		return "", err
 	}
+
+	if env == nil {
+		env = make(map[string]string)
+	}
+
+	// If customizeBuild did not set any KERNELDIR env variable,
+	// try to load kernel headers urls from driverkit.
+	if _, found := env[drivertype.KernelDirEnv]; !found {
```

Comment:
If no `customizeBuild` added the `KERNELDIR` env var to the env map, we try to autoamtically download and extract kernel headers leveraging driverkit; if we are able to do so, we emplace the new `KERNELDIR` env var pointing to the extraction path for the kernel headers.